### PR TITLE
Feat refactor presenter client

### PR DESCRIPTION
**Pull request by leungleoqin**

### CODE REVIEW

**Review by lamdbui — APPROVED** [review]
Thanks for doing the refactor, Leo. I know this was a lot of work.

**Comment by lamdbui** [comment]
I'll let @Lynne-Daniels look this over as well before we do the final merge.

**Review by Lynne-Daniels — APPROVED** [review]
Thank you!!

**Pull request merged**
